src/core/analysis/mcmc/Mcmcmc.cpp

```diff
@@ -620,6 +620,16 @@ void Mcmcmc::nextCycle(bool advanceCycle)
         if (chain_prev_boundary[i] == boundary::hottest)
             heat_visitors[heat_rank].second++;
     }
+    
+    for (size_t i = 0; i < num_chains; i++)
+    {
+        
+        if ( chains[i] != NULL )
+        {
+            setCurrentGeneration(current_generation);
```

Comment:
Running this num_chains times seems redundant?  Did you mean something like `chains[i]->setCurrentGeneration(current_generation)`?

Is this to avoid having the current generation reset to 0 for each chain?

Reply:
So for some reason, `chains[i]->setCurrentGeneration(current_generation)` did not work for me – only this did. Lemme check if I can pull it out of the loop though...

Reply:
Yes indeed, I can. Thanks @bredelings!